codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -213,6 +214,191 @@ private String generateHttpBindingSetter(Shape targetShape, Symbol targetSymbol,
+
+        // do not generate if no HTTPBinding for operation output
+        if (bindingMap.size() == 0) {
+            return;
+        }
+
+        List<String> bindingMembers = new ArrayList<>(bindingMap.keySet());
+        bindingMembers.sort(String::compareTo);
+
+        Symbol outputSymbol = symbolProvider.toSymbol(outputShape);
+        writer.addUseImports(outputSymbol);
+
+        Symbol smithyHttpResponsePointableSymbol = SymbolUtils.createPointableSymbolBuilder(
+                "Response", GoDependency.SMITHY_HTTP_TRANSPORT).build();
+        writer.addUseImports(smithyHttpResponsePointableSymbol);
+
+        writer.addUseImports(SymbolUtils.createValueSymbolBuilder(null, GoDependency.FMT).build());
+
+        String functionName = ProtocolGenerator.getOperationDeserFunctionName(outputSymbol, getProtocolName());
+        writer.openBlock("func $L(v $P, response $P) error {", "}",
+                functionName, outputSymbol, smithyHttpResponsePointableSymbol,
+                () -> {
+                    writer.openBlock("if v == nil {", "}", () -> {
+                        writer.write("return fmt.Errorf(\"unsupported deserialization for nil %T\", v)");
+                    });
+
+                    writer.write("");
+
+                    for (int i = 0; i < bindingMembers.size(); i++) {
+                        HttpBinding binding = bindingMap.get(bindingMembers.get(i));
+                        writeRestDeserializerMember(writer, model, symbolProvider, binding);
+                        writer.write("");
+                    }
+
+                    writer.write("return nil");
+                });
+    }
+
+
+    private String generateHttpBindingsValue(Shape targetShape, Symbol targetSymbol, GoWriter writer, String operand) {
+        String value = "";
+
+        Symbol strConvPackageSymbol = SymbolUtils.createValueSymbolBuilder(null, GoDependency.STRCONV).build();
+        Symbol base64EncPackageSymbol = SymbolUtils.createValueSymbolBuilder(null, GoDependency.BASE64ENCODING).build();
+        Symbol privateProtoPackageSymbol = SymbolUtils.createValueSymbolBuilder(
+                null, GoDependency.AWS_PRIVATE_PROTOCOL).build();
+
+        switch (targetShape.getType()) {
+            case STRING:
+                if (targetShape.hasTrait(EnumTrait.class)) {
+                    String typeName = "types." + targetSymbol.getName();
+                    // TODO: should this add symbol reference and import?
+                    value = typeName + "(" + operand + ")";
+                    return value;
+                }
+                return operand;
+            case BOOLEAN:
+                writer.addUseImports(strConvPackageSymbol);
+                return "strConv.ParseBool(" + operand + ")";
+            case TIMESTAMP:
+                writer.addUseImports(privateProtoPackageSymbol);
+                // Todo: needs logic to use different timestamp format for query
```

Comment:
`HttpBindingIndex` has a `determineTimestampFormat` method you should use here. Also query is an RPC protocol so it won't use this code path anyway.